src/main/java/pl/javastart/streamsexercise/PaymentService.java

```diff
@@ -19,91 +21,122 @@ class PaymentService {
     Znajdź i zwróć płatności posortowane po dacie rosnąco
      */
     List<Payment> findPaymentsSortedByDateAsc() {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll().stream().sorted(Comparator.comparing(Payment::getPaymentDate)).collect(Collectors.toList());
     }
 
     /*
     Znajdź i zwróć płatności posortowane po dacie malejąco
      */
     List<Payment> findPaymentsSortedByDateDesc() {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll().stream().sorted(Comparator.comparing(Payment::getPaymentDate).reversed()).collect(Collectors.toList());
     }
 
     /*
     Znajdź i zwróć płatności posortowane po liczbie elementów rosnąco
      */
     List<Payment> findPaymentsSortedByItemCountAsc() {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll().stream()
+                .sorted(Comparator.comparingInt(payment -> payment.getPaymentItems().size()))
+                .collect(Collectors.toList());
     }
 
     /*
     Znajdź i zwróć płatności posortowane po liczbie elementów malejąco
      */
     List<Payment> findPaymentsSortedByItemCountDesc() {
-        throw new RuntimeException("Not implemented");
+
+        return paymentRepository.findAll().stream()
+                .sorted(Comparator.comparingInt((Payment payment) -> payment.getPaymentItems().size()).reversed())
+                .collect(Collectors.toList());
     }
 
     /*
     Znajdź i zwróć płatności dla wskazanego miesiąca
      */
     List<Payment> findPaymentsForGivenMonth(YearMonth yearMonth) {
-        throw new RuntimeException("Not implemented");
+
+        return paymentRepository.findAll().stream()
+                .filter(month -> YearMonth.from(month.getPaymentDate()).equals(yearMonth)).collect(Collectors.toList());
```

Comment:
Nazwa `month` używana w tym i późniejszych metodach nie jest do końca trafna, bo Twój "month" to "payment"